Tests/SwiftFormatRulesTests/AvoidInitializersForLiteralsTests.swift

```diff
@@ -22,6 +24,8 @@ public class AvoidInitializersForLiteralsTests: DiagnosingTestCase {
                 performFunction(x: 54 as Int32)
                 performFunction(x: 54 as Int64)
                 let c = "s" as Character
+                if 3 > 2 as Int || someCondition {}
```

Comment:
Now that we're thinking about it more, I'm a bit worried about [operator precedence](https://developer.apple.com/documentation/swift/swift_standard_library/operator_declarations) here. This particular example works because `as` has higher precedence than `>`, but here's an example that wouldn't:

```swift
let x = 1...Int(5)  // fine
let x = 1...5 as Int  // error: cannot convert value of type 'ClosedRange<Int>' to type 'Int' in coercion
```

Since we can't cleanly get information about operator precedence at this stage (even if we hardcoded the built in operators, custom operators would still screw us up), we probably need to wrap the entire `foo as Bar` expression in parentheses.

Reply:
Do we want to wrap it as a single-argument tuple declaration in all cases? I would assume that we don't want `let a = (5 as Int)`. Also, I don't think we can avoid having to check the large context of the initializer literal, otherwise you could end up with `let x = 1..<((5 as Int))`.

Reply:
Yeah, we can probably try to be somewhat smart about it, by looking at parent nodes to decide if:

* the node `Int(5)` is the only thing in the expression, or
* the node `Int(5)` is surrounded immediately by parentheses

In those cases, it would be safe to not add parens of our own.

(I guess the second scenario would be represented in the syntax tree as a `TupleExpr` with one element.)

Reply:
I'm not sure what can be done for this PR since we can't reliably get the larger context of the `FunctionCallExprSyntax` node.

Reply:
Ultimately we'll need to solve this problem more generally, unless we just want to say "format rules can't access their parent node" across the board. But even that's problematic, since simple properties like `nextToken` may need to reference `parent`.

Another idea would be to have a "parenthesis reducing rule" that removes unnecessary parentheses. But we'd have to make sure that rule ran last in the pipeline, and we'd have to decide if we're happy doing it everywhere (even for parentheses that the user has added) or if we'd want to figure out a way to make it only apply to parentheses that *we* added...

The expedient thing to do right now is to just make it a lint-only rule. Let's do that, and revisit other approaches later.